lighthouse-core/test/gather/computed/first-interactive-test.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const FirstInteractive = require('../../../gather/computed/first-interactive');
+const firstInteractive = new FirstInteractive();
+const TracingProcessor = require('../../../lib/traces/tracing-processor.js');
+const Runner = require('../../../runner.js');
+
+const tooShortTrace = require('../../fixtures/traces/progressive-app.json');
+const acceptableTrace = require('../../fixtures/traces/progressive-app-m60.json');
+
+const assert = require('assert');
+
+/* eslint-env mocha */
+describe('FirstInteractive computed artifact:', () => {
+  let computedArtifacts;
+
+  beforeEach(() => {
+    computedArtifacts = Runner.instantiateComputedArtifacts();
+  });
+
+  it('throws on short traces', () => {
+    return firstInteractive.compute_({traceEvents: tooShortTrace}, computedArtifacts).then(() => {
```

Comment:
nit: for these, can this call `computedArtifacts(requestFirstInteractive({traceEvents: tooShortTrace})` so it doesn't have to call the underlying `compute_`?